hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/ListBucketHandler.java

```diff
@@ -51,7 +51,18 @@ protected void execute(OzoneClient client, OzoneAddress address)
     Iterator<? extends OzoneBucket> bucketIterator =
         vol.listBuckets(listOptions.getPrefix(), listOptions.getStartItem());
 
-    int counter = printAsJsonArray(bucketIterator, listOptions.getLimit());
+    int counter = 0;
+    printMsg("[");
+    while (bucketIterator.hasNext() && counter < listOptions.getLimit()) {
+      OzoneBucket entry = bucketIterator.next();
+      if (entry.getSourceBucket() != null && entry.getSourceVolume() != null) {
+        printObjectAsJson(new InfoBucketHandler.LinkBucket(entry));
+      } else {
+        printObjectAsJson(entry);
+      }
+      counter++;
+    }
+    printMsg("]");
```

Comment:
I don't think this is outputs a valid json list, as items are not separated by comma (`,`).

Instead of manually printing each item and trying to imitate a list, I think it would be better to replace link items with `LinkBucket` objects in-place, and let `printAsJsonArray` do the rest.